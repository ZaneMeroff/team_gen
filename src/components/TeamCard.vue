<template>
  <div class="team-card-perimeter-container">

    <p>team #{{ teamId }}</p>

    <ul>
      <li v-for="(member, index) in members" :key="index">{{ member }}</li>
    </ul>

  </div>
</template>

<script>
  export default {
    name: "TeamCard",
    props: {
      members:  { type: Array, default: () => [] },
      teamId:   { type: Number, default: 0 },
    },
  }
</script>

<style scoped>

  .team-card-perimeter-container {
    background-color: #FFF;
    border: 1px solid black;
    border-radius: 10px;
    box-shadow: 0 0 1rem rgba(110, 15, 155, .8) inset;
    height: auto;
    margin: 0px auto 20px auto;
    padding: 10px;
    width: 220px;
  }

  ul {
    margin: 4px 0px 4px 20px;
  }
</style>
