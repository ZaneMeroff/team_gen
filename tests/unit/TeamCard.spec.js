import { shallowMount } from '@vue/test-utils'

import TeamCard from "@/components/TeamCard"

describe("TeamCard", () => {

  describe("snapshots", () => {

    it("should render with default html", () => {
      const component = shallowMount(TeamCard)

      expect(component.element).toMatchSnapshot()
    })

    it("should set members prop", () => {
      const propsData = { members: [ "Bob", "Sue", "Abe" ] }
      const component = shallowMount(TeamCard, { propsData })

      expect(component.element).toMatchSnapshot()
    })

    it("should set teamId prop", () => {
      const propsData = { teamId: 5 }
      const component = shallowMount(TeamCard, { propsData })

      expect(component.element).toMatchSnapshot()
    })
  })

  describe("props", () => {

    describe("members", () => {

      it("should have a default value", () => {
        const component = shallowMount(TeamCard)

        expect(component.vm.members).toEqual([])
      })

      it("should accept an array", () => {
        const propsData = { members: [ "Bob", "Sue", "Abe" ] }
        const component = shallowMount(TeamCard, { propsData })

        expect(component.vm.members).toEqual(propsData.members)
      })
    })

    describe("teamId", () => {

      it("should have a default value", () => {
        const component = shallowMount(TeamCard)

        expect(component.vm.teamId).toEqual(0)
      })

      it("should accept a number", () => {
        const propsData = { teamId: 5 }
        const component = shallowMount(TeamCard, { propsData })

        expect(component.vm.teamId).toEqual(propsData.teamId)
      })
    })
  })
})
