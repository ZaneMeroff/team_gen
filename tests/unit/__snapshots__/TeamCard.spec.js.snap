// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamCard snapshots should render with default props 1`] = `
<div
  class="team-card-perimeter-container"
>
  <p>
    team #0
  </p>
   
  <ul />
</div>
`;

exports[`TeamCard snapshots should set members prop 1`] = `
<div
  class="team-card-perimeter-container"
>
  <p>
    team #0
  </p>
   
  <ul>
    <li>
      Bob
    </li>
    <li>
      Sue
    </li>
    <li>
      Abe
    </li>
  </ul>
</div>
`;

exports[`TeamCard snapshots should set teamId prop 1`] = `
<div
  class="team-card-perimeter-container"
>
  <p>
    team #5
  </p>
   
  <ul />
</div>
`;
